WooCommerce/Classes/ViewRelated/Upgrades/OwnerUpgradesView.swift

```diff
@@ -1,75 +1,111 @@
 import SwiftUI
+import Yosemite
+import WooFoundation
 
 struct OwnerUpgradesView: View {
-    @State var upgradePlan: WooWPComPlan
-    @State var isPurchasing = false
-    let purchasePlanAction: () -> Void
-    @State var isLoading: Bool = false
+    @State var upgradePlans: [WooWPComPlan]
+    @State var isPurchasing: Bool
+    let purchasePlanAction: (WooWPComPlan) -> Void
+    @State var isLoading: Bool
 
-    var body: some View {
-        VStack {
-            List {
-                Section {
-                    Image(upgradePlan.wooPlan.headerImageFileName)
-                        .frame(maxWidth: .infinity, alignment: .center)
-                        .listRowInsets(.zero)
-                        .listRowBackground(upgradePlan.wooPlan.headerImageCardColor)
-                        .accessibilityHidden(true)
+    init(upgradePlans: [WooWPComPlan],
+         isPurchasing: Bool = false,
+         purchasePlanAction: @escaping ((WooWPComPlan) -> Void),
+         isLoading: Bool = false) {
+        _upgradePlans = .init(initialValue: upgradePlans)
+        _isPurchasing = .init(initialValue: isPurchasing)
+        self.purchasePlanAction = purchasePlanAction
+        _isLoading = .init(initialValue: isLoading)
+    }
 
-                    VStack(alignment: .leading) {
-                        Text(upgradePlan.wooPlan.shortName)
-                            .font(.largeTitle)
-                            .accessibilityAddTraits(.isHeader)
-                        Text(upgradePlan.wooPlan.planDescription)
-                            .font(.subheadline)
-                    }
+    @State private var paymentFrequency: WooPlan.PlanFrequency = .year
+    private var paymentFrequencies: [WooPlan.PlanFrequency] = [.year, .month]
```

Comment:
At some point in the future we should make these available to the view through a viewmodel, and not expose Yosemite directly, thoughts? We currently don't have anything driving the view's data, just the closures passing data back to the `UpgradesView` when needed

Reply:
The view using Yosemite's types is fine... there's no reason that we'd need to wrap them in one part of the WooCommerce App target to hide them from another part of the same target.

Putting them on the viewModel may still be desirable, particularly the `paymentFrequencies` array, we'll see how we go with that. For this, I felt that while the user is making their selection, it's all just private view state, and once they press the button, the view model can be notified to handle the purchase.

Reply:
`_upgradePlans` references the underlying property, without going through the `@State` property wrapper, which doesn't exist until we call `State.init(initialValue:)`